x/ethbridge/keeper/keeper.go

```diff
@@ -95,32 +96,112 @@ func (k Keeper) ProcessSuccessfulClaim(ctx sdk.Context, claim string, sugaredLog
 }
 
 // ProcessBurn processes the burn of bridged coins from the given sender
-func (k Keeper) ProcessBurn(ctx sdk.Context, cosmosSender sdk.AccAddress, amount sdk.Coins, sugaredLogger *zap.SugaredLogger) error {
-	if err := k.supplyKeeper.SendCoinsFromAccountToModule(
-		ctx, cosmosSender, types.ModuleName, amount,
-	); err != nil {
-		sugaredLogger.Errorw("failed to process burn.",
+func (k Keeper) ProcessBurn(ctx sdk.Context, cosmosSender sdk.AccAddress, msg types.MsgBurn, sugaredLogger *zap.SugaredLogger) error {
+	var coins sdk.Coins
+
+	if msg.Symbol == types.CethSymbol {
+		coins = sdk.NewCoins(sdk.NewCoin(types.CethSymbol, msg.CethAmount.Add(msg.Amount)))
+	} else {
+		coins = sdk.NewCoins(sdk.NewCoin(msg.Symbol, msg.Amount), sdk.NewCoin(types.CethSymbol, msg.CethAmount))
```

Comment:
In the case that you are burning anything other than ceth, coins represents 2 different assets, ceth and the asset that you are unpegging. I think we should split these up into two separate phases, 1 burn the coin, 2, lock the ceth. 3, if there is a ceth receive account, transfer the coins from that account.

This comment is invalid if sending coins from account to module burns everything other than ceth.

Reply:
the supply keeper doesn't have the interface to transfer coin from account to account. it is more efficient to transfer both coins from account to module, then burn unpegged coin regardless the ceth as transaction fee stay at the module or go to the specific account.